zilliqa/src/consensus.rs

```diff
@@ -631,7 +626,7 @@ impl Consensus {
         let rewards_per_block =
             *self.config.consensus.rewards_per_hour / self.config.consensus.blocks_per_hour as u128;
 
-        let proposer = self.leader_at_block(parent_block, view).unwrap().public_key;
+        let proposer = self.leader_at_block(parent_block).unwrap().public_key;
```

Comment:
Shouldn't we rather use the `leaderAtView()` function of the `Deposit` contract and pass in the `view` argument? The `view` parameter of the `apply_rewards()` function could refer to an arbitrary view number, not necessarily the current view.